lstchain/io/io.py

```diff
@@ -1314,7 +1315,10 @@ def get_srcdep_params(filename, key):
         data.columns = pd.MultiIndex.from_tuples(
             [tuple(col[1:-1].replace('\'', '').replace(' ', '').split(",")) for col in data.columns])
 
-    return data[key]
+    if key is not None:
```

Comment:
if `dl1_params_src_dep_lstcam_key` and `dl2_params_src_dep_lstcam_key` are not in dataset_keys, the `data` variable does not get initialize and continuing will raise an error, maybe you could add:

```python3
else:
    raise IOError('File does not contain source-dependent parameters')
```

Reply:
ok, added